fury/ui/elements.py

```diff
@@ -3051,3 +3052,118 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+        padding : float, optional
+            Distance between loader and background.
+        bg_color : (float, float, float), optional
+            Background color of progress bar.
+        progress_color : (float, float, float), optional
+            Color of the progress bar.
+        """
+        self.bg_size = size
+        self.padding = padding
+        self.bg_color = bg_color
+        self.progress_color = progress_color
+
+        super(ProgressUI, self).__init__(position)
+
+        self.min_value = min_value
+        self.max_value = max_value
+        self.value = initial_value
+
+        # Offer some standard hooks to the user.
+        self.on_complete = lambda ui: None
+
+        self.update()
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create the ProgressUI with its background (Rectangle2D) and progress
+        (Rectangle2D).
+        """
+        self.progress_total_size = (self.bg_size[0]-2*self.padding,
+                                    self.bg_size[1]-2*self.padding)
+
+        self.background = Rectangle2D(size=self.bg_size, color=self.bg_color)
+        self.progress = Rectangle2D(size=self.progress_total_size,
+                                    color=self.progress_color)
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.background + self.loader
+
+    def _add_to_scene(self, scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        scene : Scene
+
+        """
+        self.background.add_to_scene(scene)
+        self.progress.add_to_scene(scene)
+
+    def _get_size(self):
+        return self.background.size
+
+    def _set_position(self, coords):
+        """Set the lower-left corner position of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.background.position = coords
+        self.progress.position = coords + self.padding
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        value_range = self.max_value - self.min_value
+        self.ratio = (value - self.min_value) / value_range
+
+    @property
+    def ratio(self):
+        return self._ratio
+
+    @ratio.setter
+    def ratio(self, ratio):
+        progress_width = ratio * self.progress_total_size[0]
+        if progress_width >= self.progress_total_size[0]:
+            progress_width = self.progress_total_size[0]
+            self.on_complete(self)
+        elif progress_width < 0:
+            progress_width = 0
+        self.progress.width = progress_width
+        self.update()
+
+    def update(self):
+        self._ratio = self.progress.width/self.progress_total_size[0]
+
+        value_range = self.max_value - self.min_value
+        self._value = self.min_value + self.ratio * value_range
```

Comment:
value is being set to a float instead of an integer. Hence, the value is never equal to its intended value.